pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -131,6 +131,12 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Watch for the changes happening to Infrastructure Resource
+	err = c.Watch(&source.Kind{Type: &configv1.Infrastructure{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
```

Comment:
I think sync should happen only if resource tags in the `configv1.Infrastructure` are present or updated. Do we care about the rest of the object?

Reply:
Not just watching Infrastructure, but also queueing up all the CredentialsRequests so that they get re-reconciled.

Reply:
> I think sync should happen only if resource tags in the `configv1.Infrastructure` are present on create event or modified on update event. Do we care about the rest of the object?

Not required really, I would add some predicates to check on the changes done to the ResourceTags

Reply:
> Not just watching Infrastructure, but also queueing up all the CredentialsRequests so that they get re-reconciled.

Sure, I would address this